Lombiq.NodeJs.Extensions/CustomExecTasks/SharedMutex.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Lombiq.NodeJs.Extensions.CustomExecTasks;
+
+public class SharedMutex
+    : ICustomMutex
+{
+    private readonly string _mutexName;
+    private readonly TimeSpan _timeout;
+
+    public int RetryIntervalMs { get; set; } = 100;
+
+    public SharedMutex(string mutexName, TimeSpan timeout)
+    {
+        _mutexName = mutexName;
+        _timeout = timeout;
+    }
+
+    public bool Execute(Func<bool> functionToExecute, Action<string, object[]> logWait, Action<string, object[]> logError)
+    {
+        var count = 1;
+        var stopwatch = Stopwatch.StartNew();
+        using var mutex = new Mutex(initiallyOwned: false, _mutexName);
+        while (stopwatch.Elapsed <= _timeout)
+        {
+            if (mutex.WaitOne(0))
```

Comment:
Can we use `mutex.WaitOne(RetryIntervalMs)` instead? If so, we can remove `Thread.Sleep(RetryIntervalMs);`

Reply:
No. The point is to release the mutex asap, but I've now realized that I don't actually do that!

Will refactor this code.